core/network/src/protocol.rs

```diff
@@ -652,44 +859,55 @@ impl<B: BlockT, S: NetworkSpecialization<B>, H: ExHashT> Protocol<B, S, H> {
 	}
 
 	/// Called when we propagate ready extrinsics to peers.
-	pub fn propagate_extrinsics(&self, io: &mut SyncIo) {
+	fn propagate_extrinsics(&mut self) {
 		debug!(target: "sync", "Propagating extrinsics");
 
 		// Accept transactions only when fully synced
-		if self.sync.read().status().state != SyncState::Idle {
+		if self.sync.status().state != SyncState::Idle {
 			return;
 		}
 
 		let extrinsics = self.transaction_pool.transactions();
-
+		// TODO: see if self.send_message can be made &self,
+		// so that this vec would become uncessary.
```

Comment:
1. TODO should be linked to issue
2. i don't think that's the only problem; even if `self.send_message` took `&self`, we would still be invoking it with `self.context_data.peers` borrowed mutable (not allowed). There is a free function `send_message` you can try, but it needs an `&mut peers`.

Reply:
there are a couple other similar TODOs, they should be addressed the same way.